jobs/FunctionTest/FunctionTest.groovy

```diff
@@ -140,63 +141,76 @@ def functionTest(String test_name, String label_name, String TEST_GROUP, Boolean
-                                if(fileExists ("downstream_file")) {
-                                    def props = readProperties file: "downstream_file"
-                                    failure_count = "${props.failures}".toInteger()
-                                    error_count = "${props.errors}".toInteger()
+                            try{
+                                sh '''#!/bin/bash -x
+                                set +e
+                                mkdir '''+"$artifact_dir"+'''
+                                ./build-config/post-deploy.sh
+                                files=$( ls build-deps/*.log )
+                                if [ ! -z "$files" ];then
+                                    cp build-deps/*.log '''+"$artifact_dir"+'''
+                                fi
+                                files=$( ls RackHD/test/*.xml )
+                                if [ ! -z "$files" ];then
+                                    cp RackHD/test/*.xml '''+"$artifact_dir" +'''
+                                fi
+                                if [ -d build-deps/mongodb ];then
+                                    cp -r build-deps/mongodb '''+"$artifact_dir" +'''
+                                fi
+                                '''
+                            
+                                def junitFiles = findFiles glob: 'RackHD/test/*.xml'
+                                boolean exists = junitFiles.length > 0
+                                if (exists){
+                                    junit 'RackHD/test/*.xml'
+                                    // [Based on junit xml log] Write test results to github
+                                    sh '''#!/bin/bash -x
+                                    set +e
+                                    find RackHD/test/ -maxdepth 1 -name "*.xml" > files.txt
+                                    files=$( paste -s -d ' ' files.txt )
+                                    if [ -n "$files" ];then
+                                        ./build-config/build-release-tools/application/parse_test_results.py \
+                                        --test-result-file "$files"  \
+                                        --parameters-file downstream_file
+                                    fi
+                                    '''
+                                    int failure_count = 0
+                                    int error_count = 0
+                                    if(fileExists ("downstream_file")) {
+                                        def props = readProperties file: "downstream_file"
+                                        failure_count = "${props.failures}".toInteger()
+                                        error_count = "${props.errors}".toInteger()
+                                    }
+                                    if (failure_count == 0 && error_count == 0){
+                                        result = "SUCCESS"
```

Comment:
No. if there's stopping error before test.sh runs (like vagrant up failure)
the code under ``` if(exists)``` won't be executed.
So the result is still FAILURE.